dist/cargo_test.sh

```diff
@@ -22,16 +23,52 @@ executors["cargo"]="execute_native"
 executors["docker"]="execute_docker"
 
 function run_tests() {
+  set -x
+  export CARGO_TARGET_DIR="$PWD/target"
+
+  if [[ $(type -f kcov) ]]; then
+    export HAS_KOV="${HAS_KOV:-1}"
+    rm -rf "${CARGO_TARGET_DIR}"/cov
+  fi
+
   for directory in ${!cargo_test_flags[*]}; do
+    if [[ ${HAS_KOV} -eq 1 ]]; then
+      # we want to prevent completely untested functions to be stripped
+      export RUSTFLAGS='-C link-dead-code'
+      export CARGO_ARGS="test --no-run"
+
+      # delete leftover test executibles
+      [[ ! -d "${CARGO_TARGET_DIR}"/debug ]] || find "${CARGO_TARGET_DIR}"/debug/ -maxdepth 1 -type f -executable -print -delete
+    fi
+
     (
-      set -x;
       ${1} /usr/bin/env bash -c "\
-        cd ${directory} && \
-        export CARGO_TARGET_DIR=\"../target\" && \
-        cargo ${CARGO_ARGS:-test --release} ${cargo_test_flags[${directory}]} && \
-      :"
+        cd ${directory}
+        cargo ${CARGO_ARGS:-test} ${cargo_test_flags[${directory}]}
+
+        if [[ ${HAS_KOV} -eq 1 ]]; then
+          rm -f \"${CARGO_TARGET_DIR}\"/debug/${directory}
+          find \"${CARGO_TARGET_DIR}\"/debug/ -maxdepth 1 -type f -executable -print -exec \
+          kcov \
+              --exclude-pattern=$HOME/.cargo \
+              --verify \
+              ${CARGO_TARGET_DIR}/cov \
+              {} \\;
+        fi
+      "
     )
+
   done
+
+  # TODO: remove this before merging the PR
```

Comment:
I don't see anything wrong about this for now really

Reply:
I'd prefer not to have this in by default, rather only if it's passed in by the caller. I added it to give the CI job a chance to succeed. Now that we know that it works we can properly integrate it.